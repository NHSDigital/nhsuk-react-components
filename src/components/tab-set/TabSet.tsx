/* eslint-disable @typescript-eslint/no-redeclare */
import React, { HTMLProps } from 'react';
import classNames from 'classnames';
import Tab, { TabProps } from './components/Tab';
import './_TabSet.scss';

interface TabSet extends React.FC<HTMLProps<HTMLDivElement>> {
  Tab: React.FC<TabProps>;
}

const TabSet: TabSet = ({ className, ...rest }) => (
  <div className={classNames('nhsuk-tab-set', className)} {...rest} />
);

TabSet.Tab = Tab;

export default TabSet;
