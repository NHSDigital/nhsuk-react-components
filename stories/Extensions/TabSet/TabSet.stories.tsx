/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import TabSet from '@components/tab-set/TabSet';

const meta: Meta<typeof TabSet> = {
  title: 'Extensions/TabSet',
  component: TabSet,
};
export default meta;

type Story = StoryObj<typeof TabSet>;

export const Standard: Story = {
  render: () => (
    <>
      <TabSet>
        <TabSet.Tab>Overview</TabSet.Tab>
        <TabSet.Tab active>Demographics</TabSet.Tab>
        <TabSet.Tab>Clinicals</TabSet.Tab>
        <TabSet.Tab>Documents</TabSet.Tab>
      </TabSet>
      <br />
    </>
  ),
};

export const WithDisabledTab: Story = {
  render: () => (
    <>
      <TabSet>
        <TabSet.Tab>Overview</TabSet.Tab>
        <TabSet.Tab active>Demographics</TabSet.Tab>
        <TabSet.Tab disabled>Clinicals</TabSet.Tab>
        <TabSet.Tab>Documents</TabSet.Tab>
      </TabSet>
      <br />
    </>
  ),
};

export const WithEmptyTab: Story = {
  render: () => (
    <>
      <TabSet>
        <TabSet.Tab>Overview</TabSet.Tab>
        <TabSet.Tab active>Demographics</TabSet.Tab>
        <TabSet.Tab empty>Clinicals</TabSet.Tab>
        <TabSet.Tab>Documents</TabSet.Tab>
      </TabSet>
      <br />
    </>
  ),
};

export const WithDifferentSizes: Story = {
  render: () => (
    <>
      <TabSet>
        <TabSet.Tab style={{ maxWidth: 200 }}>Overview</TabSet.Tab>
        <TabSet.Tab style={{ minWidth: 450 }}>Demographics</TabSet.Tab>
        <TabSet.Tab style={{ flexGrow: 2 }}>Clinicals</TabSet.Tab>
        <TabSet.Tab style={{ flexGrow: 0.5 }}>Documents</TabSet.Tab>
        <TabSet.Tab>Settings</TabSet.Tab>
      </TabSet>
      <br />
    </>
  ),
};
