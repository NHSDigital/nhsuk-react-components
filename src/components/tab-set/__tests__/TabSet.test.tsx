import React from 'react';
import {render} from '@testing-library/react'
import TabSet from '..';

describe('TabSet', () => {
  it('matches snapshot', () => {
    const component = render(<TabSet />)
    expect(component.container).toMatchSnapshot();
  });

  describe('Tab', () => {
    it('matches snapshot', () => {
      const component = render(<TabSet.Tab />);
      expect(component.container).toMatchSnapshot();
    });

    it('applies correct classes', () => {
      const component = render(
        <TabSet>
          <TabSet.Tab id="normal">Text</TabSet.Tab>
          <TabSet.Tab id="disabled" disabled>
            Disabled
          </TabSet.Tab>
          <TabSet.Tab id="active" active>
            Active
          </TabSet.Tab>
          <TabSet.Tab id="empty" empty>
            Empty
          </TabSet.Tab>
        </TabSet>,
      );

      expect(component.container.querySelector('#normal')?.className).toBe(
        'nhsuk-tab-set__tab')
      expect(component.container.querySelector('#disabled')?.className).toBe(
        'nhsuk-tab-set__tab nhsuk-tab-set__tab--disabled')
      expect(component.container.querySelector('#active')?.className).toBe(
        'nhsuk-tab-set__tab nhsuk-tab-set__tab--active')
      expect(component.container.querySelector('#empty')?.className).toBe(
        'nhsuk-tab-set__tab nhsuk-tab-set__tab--empty')
    });
  });
});
