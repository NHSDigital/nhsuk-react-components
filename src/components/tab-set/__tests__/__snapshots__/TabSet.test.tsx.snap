// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabSet Tab matches snapshot 1`] = `
<div>
  <button
    class="nhsuk-tab-set__tab"
    type="button"
  />
</div>
`;

exports[`TabSet matches snapshot 1`] = `
<div>
  <div
    class="nhsuk-tab-set"
  />
</div>
`;
